src/Fields/broadcast.jl

```diff
@@ -107,9 +107,8 @@ end
     space1::Type{S},
     space2::Type{S},
 ) where {S <: AbstractSpace}
-    error(
-        "Broacasted spaces are the same ClimaCore.Spaces type but not the same instance",
-    )
+    @warn "Broacasted spaces are the same ClimaCore.Spaces type but not the same instance"
+    return nothing
```

Comment:
Wouldn't we need to modify `error_mismatched_spaces` below, too? I think JET looks at all possible dispatch options when analyzing this.

Reply:
Also, we should probably change the function to `warn_mismatched_spaces` if this is the new behavior.

Reply:
> Also, we should probably change the function to `warn_mismatched_spaces` if this is the new behavior.

Yes, agreed!

> Wouldn't we need to modify error_mismatched_spaces below, too? I think JET looks at all possible dispatch options when analyzing this.

Yeah I think that is the problem we are trying to circumvent as we still want an error if the spaces aren't the same `ClimaCore.Spaces` type.

Reply:
Ah, I see. Yeah, I think we need a different (and still cheap) way of comparing spaces if we want this to pass the JET test.. Or for something in JET to change but I know less about that.